docs: clarify usage of await with useFetch

Fuzzyma opened the pull request:
Explain the pros and cons of awaiting useFetch

### 🔗 Linked issue

Well, just a discussion:
https://github.com/nuxt/nuxt/discussions/33736#discussioncomment-15046385

<!-- Please ensure there is an open issue and mention its number. For example, "resolves #123" -->

### 📚 Description

useFetch does not need to be awaited. It sometimes might even lead to better UX not to await it.
However, the docs only ever show useFetch being awaited and dont explain the implications.

Therefore I added a note explaining it a bit more.
Maybe we should put another note on useAsyncData?

<!-- Describe your changes in detail. Why is this change required? What problem does it solve? -->

<!-- If you used AI tools to help with this contribution, please ensure the PR description and code reflect your own understanding.
     Write in your own voice rather than copying AI-generated text. -->

<!----------------------------------------------------------------------
Before creating the pull request, please make sure you do the following:

- Check that there isn't already a PR that solves the problem the same way. If you find a duplicate, please help us reviewing it.
- Read the contribution docs at https://nuxt.com/docs/4.x/community/contribution
- Ensure that PR title follows conventional commits (https://www.conventionalcommits.org)
- Update the corresponding documentation if needed.
- Include relevant tests that fail without this PR but pass with it.

Thank you for contributing to Nuxt!
----------------------------------------------------------------------->

Fuzzyma committed:
Clarify usage of await with use-fetch

Explain the pros and cons of awaiting useFetch

Fuzzyma committed:
docs: fix nitpick feedback from bot

ineshbose commented:
> I would also add this note on `useAsyncData` 👍 

Yes - this note may be better suited as a section on https://nuxt.com/docs/4.x/getting-started/data-fetching and it can also tie into usage of NuxtLoadingIndicator.

Fuzzyma committed:
docs: add note to useAsyncData as well

Fuzzyma committed:
docs: add note about awaiting fetches

Added clarification on using await with useFetch and useAsyncData, discussing the impact on UX and loading states.

danielroe committed:
Merge remote-tracking branch 'origin/main' into patch-1

danielroe committed:
docs: some clarifications

autofix-ci[bot] committed:
[autofix.ci] apply automated fixes

danielroe committed:
docs: add warning about awaiting lazy functions

danielroe reviewed (approved):
thank you - this is very helpful ❤️

danielroe commented:
cc: @TheAlexLichter in view of our many conversations about this

you may also spot improvements later on after this is merged; feel free to fix then!
